apps/wallet/src/ui/app/shared/utils/ButtonOrLink.tsx

```diff
@@ -1,18 +1,29 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { type ComponentProps, forwardRef, type Ref } from 'react';
+import { type ComponentProps, forwardRef, type Ref, type ReactNode } from 'react';
 import { Link, type LinkProps } from 'react-router-dom';
 
 import LoadingIndicator from '../../components/loading/LoadingIndicator';
+import { Tooltip } from '../tooltip';
+
+type WithTooltipProps = {
+	title?: ReactNode;
+	children: ReactNode;
+};
+function WithTooltip({ title, children }: WithTooltipProps) {
```

Comment:
This really doesn't belong in this base component. It's fine because we'll eventually invest in broader refactoring work here so I think we can get away with it for now, but it really doesn't belong